wpep0008.md

```diff
@@ -0,0 +1,50 @@
+| Key | Value |
+| ---: | :--- |
+| **Title** | Wazo integration tests framework |
+| **Number** | 0008 |
+| **Date** | 2023-08-28 |
+| **Authors** | Jesse Sopel <jesse.sopel@wazo.io>, Sebastien Duthil <sduthil@wazo.io> |
+| **Status** | draft |
+
+## Abstract
+
+Wazo integration tests: Explicit the direction of evolution of integration tests
+
+## Motivation
+
+The integration test framework was implemented in Wazo before the usage of
+pytest. Pytest offers a very Pythonic way to organize tests and fixtures, and we
+want to use this system more and more, since it offers more flexibility,
+readability, performance and reusability.
+
+## State of the art in Wazo Platform
+
+Integration tests are organised around `AssetLaunchingTestCase`, inheriting from
+the `unittest` testing framework. This organization forces tests to be organized
+around a test class, which may not be as flexible as we want.
+
+## Rejected propositions
+
+* None, this proposition emerged from a refactor proposition that looked promising.
+
+## Proposition
+
+* Migrate all Wazo Platform projects to Pytest (looking at you wazo-confd!)
+* Use Pytest fixtures system as much as possible for organizing tests and fixtures.
+
+The `AssetLaunchingTestCase` class [was
+refactored](https://github.com/wazo-platform/wazo-test-helpers/pull/89) to be
+usable in more Pytest friendly patterns, while keeping retrocompatibility with
+the former style of tests.
+
+## Example
+
+An example implementation using more Pytest-oriented patterns can be read in
+[wazo-agid](https://github.com/wazo-platform/wazo-agid/pull/127/files).
+
+## Implementations
+
+A more Pytest-oriented style has been used in two repos to date:
+
```

Comment:
doesn't wazo-calld also uses mostly pytest-oriented integration tests?

Reply:
No, I don't see any pytest-based fixture in wazo-calld, and all tests rely on the `AssetLaunchingTestCase`.

Reply:
My bad, it's confd that I saw was using pytest:
https://github.com/wazo-platform/wazo-confd/blob/master/integration_tests/suite/base/test_agent_skill.py